src/db/state/record.rs

```diff
@@ -128,13 +151,29 @@ impl<'conn> State<'conn, RecordRow> {
                 }
             }
             None => {
-                self.prepare(sql::set_citation_key_ignore())?
+                self.prepare(sql::set_citation_key_fail())?
                     .execute((alias.name(), self.row_id()))?;
                 Ok(None)
             }
         }
     }
 
+    /// Check if the given alias exists and points to this row, and delete the alias if it does.
+    #[inline]
+    pub fn check_and_delete_alias(&self, alias: &Alias) -> Result<(), rusqlite::Error> {
```

Comment:
This is a bit of a confusing name, since it doesn't quite reflect that it is relative to the given row? Maybe `delete_if_has_alias` or similar?